### PR TITLE
CB-5791 Don't attempt to disable autorestart without KNOX service name

### PR DESCRIPTION
**Pull request by daszabo**

### CODE REVIEW

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerConfigService.java [hunk 1]
```diff
@@ -82,23 +85,35 @@ public void disableKnoxAutorestartIfCmVersionAtLeast(Versioned versionAtLeast, A
...
-                .findFirst().orElse(KNOX_SERVICE);
+    private Optional<String> getKnoxServiceName(String clusterName, ServicesResourceApi servicesResourceApi) {
+        try {
+            ApiServiceList serviceList = servicesResourceApi.readServices(clusterName, DataView.SUMMARY.name());
+            return serviceList.getItems().stream()
+                    .filter(service -> KNOX_SERVICE.equals(service.getType()))
+                    .map(ApiService::getName)
+                    .findFirst();
+        } catch (ApiException e) {
+            LOGGER.debug("Failed to get KNOX service name from Cloudera Manager.", e);
```
Comment (doktoric) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerConfigService.java]: Maybe the cluster name should be added to the log message

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerConfigService.java [hunk 2]
```diff
@@ -82,23 +85,35 @@ public void disableKnoxAutorestartIfCmVersionAtLeast(Versioned versionAtLeast, A
...
             ApiConfig autorestartConfig = new ApiConfig().name(KNOX_AUTORESTART_ON_STOP).value(Boolean.FALSE.toString());
             ApiServiceConfig serviceConfig = new ApiServiceConfig().addItemsItem(autorestartConfig);
-            servicesResourceApi.updateServiceConfig(clusterName, knoxServiceName, "", serviceConfig);
-        } catch (ApiException e) {
-            LOGGER.debug("Failed to set autorestart_on_stop to KNOX in Cloudera Manager.", e);
-        }
+            try {
+                servicesResourceApi.updateServiceConfig(clusterName, knoxServiceName, "", serviceConfig);
+            } catch (ApiException e) {
+                LOGGER.debug("Failed to set autorestart_on_stop to KNOX in Cloudera Manager.", e);
```
Comment (doktoric) [inline cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerConfigService.java]: Maybe the cluster name should be added to the log message

**Comment by doktoric** [comment]
I think unit test should be useful to cover these new lines in the code

File: cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerConfigService.java (hunk 1 above)
Comment (daszabo) [reply cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerConfigService.java]: Ideally the log entry should already contain it.

**Pull request merged**